### DISCUSSION TITLE
Node in core emulator system is unable to communicate with outside physical system using RJ45

### DISCUSSION THREAD

**Discussion by meenakshi510** [coreemu/core#631] [Q&A] [answered]
![core emulator](https://user-images.githubusercontent.com/91558188/139585501-f61bc924-ae73-466f-823c-376d93fd3adf.png)

Node n2 is communicating with outside physical system but node n1 is not able to communicate with outside physical system.
Kindly help.

**Comment by gh0st42** [comment, marked as answer]
n1 (10.0.0.0/24) is in a different subnet than n2 and also the physical eth0 (10.0.1.0/24)

So the physical node needs either its default gw to be on 10.0.1.1/24 or have a net route for 10.0.0.0/24 via 10.0.1.1
n1 then also need a route back via 10.0.0.2 for the 10.0.1.0/24 subnet. 

If both have the extra route entries, everything should work.

**Reply by meenakshi510** [reply]
@gh0st42 I had kept the default gw address of the physical node from the same subnet 10.0.1.0/24. But it didn't work.

How to create the extra route entries? Please suggest.

**Reply by gh0st42** [reply]
Depends on your OS, on Linux something like `ip route 10.0.0.0/24 via 10.0.1.1` or `route add -net 10.0.0.0/24 gw 10.0.1.1` should do the trick. 
Beware: These are temporary static routes and are gone after a reboot.

**Reply by meenakshi510** [reply]
@gh0st42  If we need to add more intermediate nodes between n1 and n2 then should we need to add all those addresses in ip route?

**Reply by gh0st42** [reply]
you need routes for all the subnets you want to communicate with and that are not local to your node. 
for more complex setups you might want to consider a (mesh/classic) routing daemon.

**Reply by meenakshi510** [reply]
@gh0st42 Thank you so much I got an idea.
